src/components/candevice/candevicemodel.cpp

```diff
@@ -0,0 +1,98 @@
+#include "candevicemodel.h"
+#include <QtCore/QDir>
+#include <QtCore/QEvent>
+
+#include <QtWidgets/QFileDialog>
+
+#include <nodes/DataModelRegistry>
+
+#include "datamodeltypes/candevicedata.h"
+
+CanDeviceModel::CanDeviceModel()
+    : label(new QLabel())
+{
+    label->setAlignment(Qt::AlignVCenter | Qt::AlignHCenter);
+
+    label->setFixedSize(75, 25);
+
+    label->installEventFilter(this);
+
+    label->setAttribute(Qt::WA_TranslucentBackground);
+
+    canDevice = new CanDevice(factory);
+
+    connect(canDevice, &CanDevice::frameSent, this, &CanDeviceModel::frameSent);
+    connect(canDevice, &CanDevice::frameReceived, this, &CanDeviceModel::frameReceived);
+
+    canDevice->init("socketcan", "can0");
+    canDevice->start();
+}
+
+CanDeviceModel::~CanDeviceModel() { delete canDevice; }
```

Comment:
Is there any reason against using `unique_ptr` instead of a naked new-delete pair?